### PR TITLE
Fix ChibiOS USB config for chips which support OTGv1

### LINKED ISSUE #4783: STM32 with OTG support has differently named USBEndpointConfig struct member
**Describe the bug**
STM32 chips with USB OTG support apparently use the OTGv1 HAL code instead of the USBv1 code. In hal_usb_lld.h for OTGv1, the USBEndpointConfig struct has an in_multiplier field:
```c
/**
 * @brief   Determines the space allocated for the TXFIFO as multiples of
 *          the packet size (@p in_maxsize). Note that zero is interpreted
 *          as one for simplicity and robustness.
 */
uint16_t in_multiplier;
```
instead of USBv1's ep_buffers field:
```c
/**
 * @brief   Reserved field, not currently used.
 * @note    Initialize this field to 1 in order to be forward compatible.
 */
uint16_t ep_buffers;
```

The problem occurs in tmk_core/protocol/chibios/usb_main.c where the QMK_USB_DRIVER_CONFIG macro defines 3 USBEndpointConfigs, all with `.ep_buffers = 2,`. When compiling for my STM32F4 Discovery board, I get an error saying that USBEndpointConfig has no member named ep_buffers.

**System Information**
 - Keyboard:  STM32F4 Discovery board
   - Revision (if applicable):  
 - Operating System: Linux
 - avr-gcc version: 8.2.0

 - arm gcc version: 8.2.0

 - QMK Firmware version: 0.6.222

 - Any keyboard related software installed? 
   - [ ] Auto Hot Key
   - [ ] Karabiner
   - [ ] Other

**Additional context**

### PR DESCRIPTION
**Pull request by zvecr**
## Description

As per #4783 USBv1 devices have `ep_buffers` and OTGv1 have `in_multiplier`.

This PR aligns usb endpoint config as per rest of file, using explicit order instead of struct member name.

## Types of Changes

- [x] Core
- [x] Bugfix
- [ ] New feature
- [ ] Enhancement/optimization
- [ ] Keyboard (addition or update)
- [ ] Keymap/layout/userspace (addition or update)
- [ ] Documentation

## Issues Fixed or Closed by This PR

* Fixes #4783

## Checklist

- [x] My code follows the code style of this project.
- [ ] My change requires a change to the documentation.
- [ ] I have updated the documentation accordingly.
- [x] I have read the [**CONTRIBUTING** document](https://docs.qmk.fm/#/contributing).
- [ ] I have added tests to cover my changes.
- [x] I have tested the changes and verified that they work and don't break anything (as well as I can manage).

### CODE REVIEW

**Comment by zvecr** [comment]
The rest of the file does the whole implicit order, I like the idea of adding comments, but unsure of where is best to add. Should i do it to all sites of the usb config stuff? Just at the top means it would potentially get missed.

Also, with #7563 merged we will see compilation issues so should know if future work breaks something.

**Comment by yanfali** [comment]
> The rest of the file does the whole implicit order, I like the idea of adding comments, but unsure of where is best to add. Should i do it to all sites of the usb config stuff? Just at the top means it would potentially get missed.
> 

Maybe a major comment at top, and then a small see comment at top with all the sites you think it's relevant?

File: tmk_core/protocol/chibios/usb_main.c [hunk 1]
```diff
@@ -15,6 +15,16 @@
  * GPL v2 or later.
  */
 
+/*
+ * Implementation notes:
+ *
+ * USBEndpointConfig - Configured using explicit order instead of struct member name.
+ *   This is due to ChibiOS hal LDD differences, which is dependent on hardware,
```
Comment (fauxpark) [inline tmk_core/protocol/chibios/usb_main.c]:
```suggestion
 *   This is due to ChibiOS hal LLD differences, which is dependent on hardware,
```

Comment (zvecr) [reply tmk_core/protocol/chibios/usb_main.c]: Cheers!

**Pull request merged**